plugins/00_dokku-standard/commands

```diff
@@ -3,6 +3,76 @@ set -eo pipefail; [[ $DOKKU_TRACE ]] && set -x
 source "$(dirname $0)/../common/functions"
 
 case "$1" in
+  build)
+    APP="$2"; IMAGE="dokku/$APP"; IMAGE_SOURCE_TYPE="$3"; TMP_WORK_DIR="$4"
+    CACHE_DIR="$DOKKU_ROOT/$APP/cache"
+
+    pushd "$TMP_WORK_DIR" &> /dev/null
+
+    case "$IMAGE_SOURCE_TYPE" in
+      buildstep)
+        id=$(tar -c . | docker run -i -a stdin $DOKKU_IMAGE /bin/bash -c "mkdir -p /app && tar -xC /app")
+        test "$(docker wait $id)" -eq 0
+        docker commit $id $IMAGE > /dev/null
+        [[ -d $CACHE_DIR ]] || mkdir $CACHE_DIR
+        pluginhook pre-build "$APP"
+
+        # *DEPRECATED* in v0.4.0: `pluginhook docker-args` will be removed in future releases
+        # https://github.com/progrium/dokku/issues/896 & https://github.com/progrium/dokku/issues/906
+        DOCKER_ARGS=$(: | pluginhook docker-args $APP build)
+        DOCKER_ARGS+=$(: | pluginhook docker-args-build $APP)
+        id=$(docker run -d -v $CACHE_DIR:/cache -e CACHE_PATH=/cache $DOCKER_ARGS $IMAGE /build/builder)
+        docker attach $id
+        test "$(docker wait $id)" -eq 0
+        docker commit $id $IMAGE > /dev/null
+
+        pluginhook post-build "$APP"
```

Comment:
So like this?

post-build-buildstep:
```
exec our stuff....
pluginhook post-build "$APP"
```
That's a different approach than our previous pluginhook deprecations. Should we implement yet another?